File: app/src/main/java/com/example/todo/data/TodoDao.java
package com.example.todo.data;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

@Dao
public interface TodoDao {
    @Insert
    long insertTodo(TodoItem item);

    @Update
    void updateTodo(TodoItem item);

    @Delete
    void deleteTodo(TodoItem item);

    @Query("select * from TodoItem where created = 0")
    List<TodoItem> loadAllTodoItem();

    @Query("select * from TodoItem where type in (:arg1, :arg2) order by type")
    List<TodoItem> loadNowTodoItem(int arg1, int arg2);

    @Query("select * from TodoItem where type = :type and created = 0 order by type")
    List<TodoItem> loadSpecifiedTypeTodoItem(int type);

    @Query("select * from TodoItem where id = :id")
    TodoItem loadSpecifiedIdTodoItem(long id);
}

File: app/src/main/java/com/example/todo/data/TodoDao.kt
package com.example.todo.data

import androidx.room.*
import com.example.todo.utils.TodoTypeUtil

@Dao
interface TodoDao {
    @Insert
    fun insertTodo(item: TodoItem): Long

    @Update
    fun updateTodo(item: TodoItem)

    @Delete
    fun deleteTodo(item: TodoItem)

    @Query("select * from TodoItem where periodDone = 0")
    fun loadAllTodoItem(): List<TodoItem>

    @Query("select * from TodoItem where type not in (:arg1) order by type")
    fun loadNowTodoItem(arg1: Int = TodoTypeUtil.TYPE_LATER): MutableList<TodoItem>

    @Query("select * from TodoItem where type not in (:arg1, :arg2) order by type")
    fun loadNormalAndUrgentItem(
        arg1: Int = TodoTypeUtil.TYPE_URGENT,
        arg2: Int = TodoTypeUtil.TYPE_NORMAL
    ): MutableList<TodoItem>

    @Query("select * from TodoItem where type = :type")
    fun loadSpecifiedTypeTodoItem(type: Int): List<TodoItem>

    @Query("select * from TodoItem where periodDone = 1")
    fun loadPeriodDoneTodoItem(): List<TodoItem>

    @Query("select * from TodoItem where id = :id")
    fun loadSpecifiedIdTodoItem(id: Long): TodoItem
}